core/cas-server-core-events/src/test/java/org/apereo/cas/support/events/CasAuthenticationEventListenerTests.java

```diff
@@ -79,19 +94,23 @@ public void verifyCasAuthenticationWithNoClientInfo() {
         val event = new CasAuthenticationTransactionFailureEvent(this,
             CollectionUtils.wrap("error", new FailedLoginException()),
             CollectionUtils.wrap(CoreAuthenticationTestUtils.getCredentialsWithSameUsernameAndPassword()));
+        var current = numEventsReceived.get();
         applicationContext.publishEvent(event);
+        sleep(current + 1);
```

Comment:
This is not a good idea and should be avoided in almost all cases. You can use `await()`

Reply:
I refactored this a bit.  These are calling either publishEventAndWaitToProcess or waitForSpringEventToProcess, and is now using await.